src/transformStateWithClones.js

```diff
@@ -7,7 +7,53 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stagesOfModify = [];
+  const modifyingState = { ...state };
+
+  for (const action of actions) {
+    const { type, ...rest } = action;
+    let modifiedObject = {};
+
+    switch (type) {
+      case 'clear':
+        modifiedObject = clear(modifyingState);
+        break;
+      case 'removeProperties':
+        modifiedObject = removeProperties(modifyingState, rest);
```

Comment:
Instead of passing 'rest' to the removeProperties function, pass 'rest.keysToRemove'.